TLSharp.Core/TelegramClient.cs

```diff
@@ -406,6 +414,221 @@ public async Task<TLAbsMessages> GetHistoryAsync(TLAbsInputPeer peer, int offset
+        /// Values smaller than 0 are ignored. If stIdx is set, a number of results smaller than pageSize might be returned by Telegram.</param>
+        /// <param name="partType">The type of the participants to get. Choose Recents not to filter</param>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        public async Task<TLChannelParticipants> GetParticipants(TLChannel channel, int stIdx = -1, int pageSize = -1, ParticipantTypes partType = ParticipantTypes.Recents, CancellationToken token = default(CancellationToken))
+        {
+            if (channel == null) return null;
+            return await GetParticipants(channel.Id, (long)channel.AccessHash, stIdx, pageSize, partType, token).ConfigureAwait(false);
+        }
+        /// <summary>
+        /// Gets the participants of the channel having the supplied type.
+        /// The method will auto-paginate results and return all the participants
+        /// </summary>
+        /// <param name="channelId">The id of the channel whose participants are requested</param>
+        /// <param name="accessHash">The access hash of the channel whose participants are requested</param>
+        /// <param name="stIdx">The index to start fetching from. -1 will automatically fetch all the results</param>
+        /// <param name="pageSize">How many results to be fetch on each iteration. 
+        /// Values smaller than 0 are ignored. If stIdx is set, a number of results smaller than pageSize might be returned by Telegram.</param>
+        /// <param name="partType">The type of the participants to get. Choose Recents not to filter</param>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        public async Task<TLChannelParticipants> GetParticipants(int channelId, long accessHash, int stIdx = -1, int pageSize = -1, ParticipantTypes partType = ParticipantTypes.Recents, CancellationToken token = default(CancellationToken))
+        {
+            TLAbsChannelParticipantsFilter filter;
+            switch (partType)
+            {
+                case ParticipantTypes.Admins:
+                    filter = new TLChannelParticipantsAdmins();
+                    break;
+
+                case ParticipantTypes.Kicked:
+                    filter = new TLChannelParticipantsKicked();
+                    break;
+
+                case ParticipantTypes.Bots:
+                    filter = new TLChannelParticipantsBots();
+                    break;
+
+                case ParticipantTypes.Recents:
+                    filter = new TLChannelParticipantsRecent();
+                    break;
+
+                case ParticipantTypes.Banned:
+                case ParticipantTypes.Restricted:
+                case ParticipantTypes.Contacts:
+                case ParticipantTypes.Search:
+                default:
+                    throw new NotImplementedException($"{partType} not implemented yet");
```

Comment:
If it's not gonna be implemented please remove them from your enum as the only purpose of enum is for this filter

Reply:
it's gonna be implemented as soon as the new Classes are added to the TL framework

Reply:
should i drop them anyways or leave them?

Reply:
are they present in the new layers?

Reply:
yes, see here: https://core.telegram.org/type/ChannelParticipantsFilter

Reply:
I vote, then, for only merging this PR after we have updated the layer